lib/extended-css-selector.js

```diff
@@ -284,25 +285,11 @@ var ExtendedSelector = (function () { // jshint ignore:line
         if (!simpleNodes || !simpleNodes.length) {
             return resultNodes;
         }
-
-        var iSimpleNodes = simpleNodes.length;
+        var iSimpleNodes = simpleNodes.length; 
         while (iSimpleNodes--) {
             var node = simpleNodes[iSimpleNodes];
-            var childNodes = Sizzle(compiledSelector.complex, node); // jshint ignore:line
-            if (compiledSelector.relation === ">") {
-                // Filter direct children
-                var iChildNodes = childNodes.length;
-                while (iChildNodes--) {
-                    var childNode = childNodes[iChildNodes];
-                    if (childNode.parentNode === node) {
-                        resultNodes.push(childNode);
-                    }
-                }
-            } else {
-                resultNodes = resultNodes.concat(childNodes);
-            }
+            Sizzle(compiledSelector.relation + compiledSelector.complex, node, resultNodes); // jshint ignore:line
```

Comment:
As I recall, `Sizzle` becomes very slow when you add a relation token at the beginning of the selector.

Just compare the old and the new one's performance.

Reply:
Ok, but there is no other correct way. In order to do other optimizations, we may need to narrow down further selectors that we optimize for.

Reply:
Yep, you're right

Reply:
However, I'd like to experiment to find out if further optimization is possible by narrowing down the scope of selectors that we optimize

Reply:
Bump the ExtCss version once it's done.

Frankly, I suppose this solution is not bad already.